From 40cd5af7e75fbb677ae5c725f63ddf282e143a04 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 17 Sep 2024 08:05:50 +0200
Subject: [PATCH 1258/1695] drivers: stm32_ltdc: apply secure pinctrl

Apply pinctrl when LTDC is loaded in secure mode.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c07fe20aa8a34f85346eb17d254c9efb7bf1b4f3 drivers: stm32_ltdc: apply secure pinctrl]
Change-Id: Ib913de2078f05135ebe843b4325f9190511d4899
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/404873
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/stm32_ltdc.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/core/drivers/stm32_ltdc.c b/core/drivers/stm32_ltdc.c
index d0bdb7d11..94f4ab975 100644
--- a/core/drivers/stm32_ltdc.c
+++ b/core/drivers/stm32_ltdc.c
@@ -162,6 +162,14 @@ static TEE_Result stm32_ltdc_init(void *device)
 	if (ret)
 		goto err;
 
+	if (ldev->pinctrl) {
+		ret = pinctrl_apply_state(ldev->pinctrl);
+		if (ret) {
+			firewall_set_alternate_conf(ldev->nsec_conf);
+			goto err;
+		}
+	}
+
 	return TEE_SUCCESS;
 err:
 	clk_disable(ldev->clock);
@@ -473,12 +481,6 @@ static TEE_Result stm32_ltdc_probe(const void *fdt, int node,
 	if (ret && ret != TEE_ERROR_ITEM_NOT_FOUND)
 		goto err3;
 
-	if (ldev->pinctrl) {
-		ret = pinctrl_apply_state(ldev->pinctrl);
-		if (ret)
-			goto err4;
-	}
-
 	ret = firewall_dt_get_alternate_conf(fdt, node, "sec",
 					     &ldev->sec_conf);
 	if (ret)
-- 
2.47.2

